packages/core/src/SessionEventEmitter.ts

```diff
@@ -0,0 +1,256 @@
+   * @param eventName The session restoration event name.
+   * @param listener The callback called on the next successful session restore.
+   */
+  once(
+    eventName: typeof EVENTS.SESSION_RESTORED,
+    listener: (currentUrl: string) => unknown
+  ): this;
+  /**
+   * Register a listener called on the next error, with an error identifier and description.
+   * @param eventName The error event name.
+   * @param listener The callback called on the next error.
+   */
+  once(
+    eventName: typeof EVENTS.ERROR,
+    listener: (
+      error: string | null,
+      errorDescription?: string | null
+    ) => unknown
+  ): this;
+  /**
+   * Register a listener called on the next session extension.
+   * @param eventName The session extension event name.
+   * @param listener The callback called on the next session extension.
+   */
+  once(
+    eventName: typeof EVENTS.SESSION_EXTENDED,
+    listener: (expiresIn: number) => void
+  ): this;
+  /**
+   * Register a listener called the next time a timeout is set for a session event
+   * with the timeout handle.
+   * @param eventName The timeout set event name.
+   * @param listener The callback called when next setting a timeout.
+   */
+  once(
+    eventName: typeof EVENTS.TIMEOUT_SET,
+    listener: (timeoutHandle: number) => void
+  ): this;
+  /**
+   * Register a listener called the next time a new refresh token is issued for
+   * the session.
+   * @param eventName The new refresh token issued event name.
+   * @param listener The callback called next time a new refresh token is issued.
+   */
+  once(
+    eventName: typeof EVENTS.NEW_REFRESH_TOKEN,
+    listener: (newToken: string) => void
+  ): this;
+  once(
+    eventName: Parameters<InstanceType<typeof EventEmitter>["on"]>[0],
+    listener: Parameters<InstanceType<typeof EventEmitter>["on"]>[1]
+  ): this {
+    return super.once(eventName, listener);
+  }
+
+  /**
+   * Unegister a listener called on successful login with the logged in WebID.
+   * @param eventName The login event name.
+   * @param listener The callback to unregister.
+   */
+  off(eventName: typeof EVENTS.LOGIN, listener: (webid: string) => void): this;
+  /**
+   * Unegister a listener called on successful logout.
+   * @param eventName The logout event name.
+   * @param listener The callback to unregister.
+   */
+  off(eventName: typeof EVENTS.LOGOUT, listener: () => void): this;
+  /**
+   * Unegister a listener called on session expiration.
+   * @param eventName The session expiration event name.
+   * @param listener The callback to unregister.
+   */
+  off(eventName: typeof EVENTS.SESSION_EXPIRED, listener: () => void): this;
+  /**
+   * Unegister a listener called on session restoration after a silent login.
+   * @param eventName The session restoration event name.
+   * @param listener The callback to unregister.
+   */
+  off(
+    eventName: typeof EVENTS.SESSION_RESTORED,
+    listener: (currentUrl: string) => unknown
+  ): this;
+  /**
```

Comment:
Most of the `off` handlers appear to have an incorrect description for the `listener`

Reply:
Do you mean in the type description or the API docs? Based on https://nodejs.org/api/events.html#emitterremovelistenereventname-listener the API docs sounds right, although I'm open to any change of phrasing. I'll double-check the type signatures.